ioctl/cmd/alias/aliasimport.go

```diff
@@ -24,58 +26,83 @@ var aliasImportCmd = &cobra.Command{
 	Args:  cobra.ExactArgs(1),
 	RunE: func(cmd *cobra.Command, args []string) error {
 		cmd.SilenceUsage = true
-		output, err := aliasImport(cmd, args)
-		if err == nil {
-			println(output)
-		}
+		err := aliasImport(cmd, args)
 		return err
 	},
 }
 
+type importMessage struct {
+	ImportedNumber int     `json:"importedNumber"`
+	TotalNumber    int     `json:"totalNumber"`
+	Imported       []alias `json:"imported"`
+	Unimported     []alias `json:"unimported"`
+}
+
 func init() {
 	aliasImportCmd.Flags().StringVarP(&format,
 		"format=", "f", "json", "set format: json/yaml")
 	aliasImportCmd.Flags().BoolVarP(&forceImport,
 		"force-import", "F", false, "cover existed aliases forcely")
 }
 
-func aliasImport(cmd *cobra.Command, args []string) (string, error) {
+func aliasImport(cmd *cobra.Command, args []string) error {
 	var importedAliases aliases
-
 	switch format {
 	default:
-		cmd.SilenceUsage = false
-		return "", fmt.Errorf("invalid flag %s", format)
+		return output.PrintError(output.FlagError, fmt.Sprintf("invalid format flag %s", format))
 	case "json":
 		if err := json.Unmarshal([]byte(args[0]), &importedAliases); err != nil {
-			return "", err
+			return output.PrintError(output.SerializationError, err.Error())
 		}
 	case "yaml":
 		if err := yaml.Unmarshal([]byte(args[0]), &importedAliases); err != nil {
-			return "", err
+			return output.PrintError(output.SerializationError, err.Error())
 		}
 	}
 	importedNum, totalNum := 0, 0
 	aliases := GetAliasMap()
+	importMessage := importMessage{TotalNumber: len(importedAliases.Aliases), ImportedNumber: 0}
 	for _, importedAlias := range importedAliases.Aliases {
 		totalNum++
 		if !forceImport && config.ReadConfig.Aliases[importedAlias.Name] != "" {
-			fmt.Println("existed alias " + importedAlias.Name)
+			importMessage.Unimported = append(importMessage.Unimported, importedAlias)
+			importMessage.ImportedNumber++
 			continue
 		}
 		for aliases[importedAlias.Address] != "" {
-			delete(config.ReadConfig.Aliases, aliases[importedAlias.Name])
+			delete(config.ReadConfig.Aliases, aliases[importedAlias.Address])
 			aliases = GetAliasMap()
 		}
 		config.ReadConfig.Aliases[importedAlias.Name] = importedAlias.Address
+		importMessage.Imported = append(importMessage.Imported, importedAlias)
 		importedNum++
 	}
 	out, err := yaml.Marshal(&config.ReadConfig)
 	if err != nil {
-		return "", err
+		return output.PrintError(output.SerializationError, err.Error())
 	}
 	if err := ioutil.WriteFile(config.DefaultConfigFile, out, 0600); err != nil {
-		return "", fmt.Errorf("failed to write to config file %s", config.DefaultConfigFile)
+		return output.PrintError(output.WriteFileError,
+			fmt.Sprintf("failed to write to config file %s", config.DefaultConfigFile))
+	}
+	printImport(importMessage)
+	return nil
+}
+
+func printImport(message importMessage) {
```

Comment:
I recommend you define a `String(format string)` interface for Message as well as a method for Output. Then . you implement the interface for all kinds of messages, which will make your code look cleaner

Reply:
Cool, I will do that.